graphstorm-processing/graphstorm_processing/data_transformations/dist_transformations/base_dist_transformation.py

```diff
@@ -52,6 +62,28 @@ def get_json_representation(self) -> dict:
         else:
             return {}
 
+    def apply_precomputed_transformation(self, input_df: DataFrame) -> DataFrame:
```

Comment:
This function acts as a fail-safe, in case somehow we try to re-apply a transformation for a transformer that doesn't support re-applying.

This can only happen if a `precomputed_transformations.json` is modified by hand to include a value for a not-supported transformation.

The alternative is to raise a `NotImplementedError()` here, but I think this is safer.

Reply:
I did not fully understand the point here. I think we want to avoid to run the not-supported transformation when re-applying the transformation, right? But in this way we are still running the previous apply function on not-supported transformation, but `NotImplementedError()` will stop the whole process. Why will the method be safer?

Reply:
The idea here is to have is do apply a new transformation instead of errorring out, in the convoluted even that a non-supported transformation ends up in `precomputed_transformations.json`. The scenario is the following:


* A user runs gs-processing on some data, let's say they have a numerical feature, `age`, and a categorical feature, `state`.
* We save the transformation for `state` in `precomputed_transformations.json` under the output path.
* The user modifies `precomputed_transformations.json` to include the `age` feature, and copies the file to some new input data.
* When the user runs gs-processing on the new data, we'll read an entry for `age` in `precomputed_transformations.json`, and will call `DistNumericalTransformation.apply_precomputed_transformation()` to re-apply it.
* The default implementation will run here, printing a warning and creating a new transformation.

I realize though that the scenario that "The user modifies `precomputed_transformations.json` to include the `age` feature" is very unlikely. Maybe just having `NotImplementedError()` here is better? That way if a non-supported feature ends up in `precomputed_transformations.json` for any reason we just error out.

Reply:
That makes sense.